packages/foam-vscode/src/features/preview-navigation.ts

```diff
@@ -96,9 +96,13 @@ export const markdownItWithRemoveLinkReferences = (
   md: markdownit,
   workspace: FoamWorkspace
 ) => {
-  // Forget about reference blocks before processing links.
+  // Forget about reference links that contain an alias divider
   md.inline.ruler.before('link', 'clear-references', state => {
-    state.env.references = undefined;
+    Object.keys(state.env.references).map(refKey => {
```

Comment:
minor, but let's make linter happy:

```suggestion
    Object.keys(state.env.references).forEach(refKey => {
```